### PR TITLE
CLI args override values in JSON file

### PR DESCRIPTION
**Pull request by damian0815**
If `--config config.json` is passed, any additional CLI args will now override the values in config.json.

eg if config.json is `{"resume_ckpt": "sd21.ckpt", "seed": 555}` then running `python train.py --config.json --seed 123 --lr 1e-6` will result in `args.resume_ckpt == "sd21.ckpt"` and `args.seed == 123` and `args.lr == 1e-6`.

if resume_ckpt is omitted from `config.json`, it will be required as a CLI arg, otherwise it will be optional.

(credit to @noprompt for the implementation hint)

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -995,62 +995,63 @@ def update_old_args(t_args):
...
+    argparser.add_argument("--lr_scheduler", type=str, default="constant", help="LR scheduler, (default: constant)", choices=["constant", "linear", "cosine", "polynomial"])
+    argparser.add_argument("--lr_warmup_steps", type=int, default=None, help="Steps to reach max LR during warmup (def: 0.02 of lr_decay_steps), non-functional for constant")
+    argparser.add_argument("--max_epochs", type=int, default=300, help="Maximum number of epochs to train for")
+    argparser.add_argument("--notebook", action="store_true", default=False, help="disable keypresses and uses tqdm.notebook for jupyter notebook (def: False)")
+    argparser.add_argument("--project_name", type=str, default="myproj", help="Project name for logs and checkpoints, ex. 'tedbennett', 'superduperV1'")
+    argparser.add_argument("--resolution", type=int, default=512, help="resolution to train", choices=supported_resolutions)
+    argparser.add_argument("--resume_ckpt", type=str, required=not ('resume_ckpt' in args), default="sd_v1-5_vae.ckpt", help="The checkpoint to resume from, either a local .ckpt file, a converted Diffusers format folder, or a Huggingface.co repo id such as stabilityai/stable-diffusion-2-1 ")
```
Comment (damian0815) [inline train.py:1036]: this is the only change in this block (`required=not ('resume_ckpt' in args)`) - easier to see with `git diff -w` to ignore whitespace

File: train.py [hunk 2]
```diff
@@ -995,62 +995,63 @@ def update_old_args(t_args):
...
     if args.config is not None:
-        print(f"Loading training config from {args.config}, all other command options will be ignored!")
+        print(f"Loading training config from {args.config}.")
         with open(args.config, 'rt') as f:
-            t_args = argparse.Namespace()
-            t_args.__dict__.update(json.load(f))
-            update_old_args(t_args) # update args to support older configs
-            args = argparser.parse_args(namespace=t_args)
+            args.__dict__.update(json.load(f))
+            update_old_args(args) # update args to support older configs
```
Comment (noprompt) [inline train.py:1004]: With this change `update_old_args` is no longer necessary since the second pass over the arguments will apply the same defaults:

* `shuffle_tags` defaults to `False`,
* `save_full_precision` defaults to `False`,
* `notebook` defaults to `False`,
* `disable_unet_training` defaults to `False`,
* `rated_dataset` defaults to `False`, and
* `rated_dataset_target_dropout_percent` defaults to `50`.

These are the same defaults provided by `update_old_args`. 😄

Comment (damian0815) [reply train.py:1004]: ahha awesome. thanks for taking a look!

Comment (damian0815) [reply train.py]: i removed the function, thanks for the tip
